plugins/modules/omevv_firmware_repository_profile.py

```diff
@@ -249,19 +272,26 @@ def create_firmware_repository_profile(self):
                 share_domain=self.module.params.get('share_domain')
             )
             if resp.success:
-                profile_resp = self.omevv_profile_obj.get_firmware_repository_profile_by_id(resp.json_data)
-                while profile_resp.json_data["status"] != "Success" and profile_resp.json_data["status"] != "Failed":
-                    time.sleep(3)
-                    profile_resp = self.omevv_profile_obj.get_firmware_repository_profile_by_id(resp.json_data)
-                if self.module._diff and profile_resp.json_data["status"] == "Success":
-                    self.module.exit_json(msg=SUCCESS_CREATION_MSG, profile_info=profile_resp.json_data, diff=diff, changed=True)
-                elif profile_resp.json_data["status"] == "Success":
-                    self.module.exit_json(msg=SUCCESS_CREATION_MSG, profile_info=profile_resp.json_data, changed=True)
-                else:
-                    self.module.exit_json(msg=FAILED_CREATION_MSG, profile_info=profile_resp.json_data, failed=True)
+                self.diff_mode_behaviour(resp)
             else:
                 self.module.exit_json(msg=FAILED_CREATION_MSG, failed=True)
 
+    def diff_mode_behaviour(self, resp):
+        profile_resp = self.omevv_profile_obj.get_firmware_repository_profile_by_id(resp.json_data)
+        while profile_resp.json_data["status"] != "Success" and profile_resp.json_data["status"] != "Failed":
+            time.sleep(3)
```

Comment:
Why are we waiting for 3 sec here?

Reply:
On every 3 sec it will do a GET call to check whether profile creation is successfull or not.